BlueDBOnDisk/src/test/java/org/bluedb/disk/file/FileManagerTest.java

```diff
@@ -88,7 +94,82 @@ public void test_saveObject() throws Exception {
 		Object reloadedNull = fileManager.loadObject(fileWithNull.toPath());
 		assertNull(reloadedNull);
 	}
+	
+	@Test
+	public void test_saveAndLoadVersionedObjects() throws Exception {
+		File tempDir = createTempFolder("test_saveVersionedObject");
+		Path tempDirPath = tempDir.toPath();
+		String filename = "test_filename";
+		
+		TestValue value1 = new TestValue("Joe Dirt", 5);
+		Path value1Path = tempDirPath.resolve(filename);
+		
+		fileManager.saveObject(value1Path, value1);
+		
+		assertEquals(value1, fileManager.loadVersionedObject(tempDirPath, filename)); //It should load the legacy non-versioned object if its the only one
+		FileTime value1LastModified = Files.getLastModifiedTime(value1Path);
 
+		
+		TestValue value2 = new TestValue("Michael Scott", 3);
+		fileManager.saveVersionedObject(tempDirPath, filename, value2);
+		
+		Path value2Path = FileManager.getNewestVersionPath(tempDirPath, filename);
+		FileTime value2LastModified = Files.getLastModifiedTime(value2Path);
+		
+		assertEquals(value1, fileManager.loadObject(value1Path));
+		assertEquals(value1LastModified, Files.getLastModifiedTime(value1Path));
+		assertEquals(value2, fileManager.loadVersionedObject(tempDirPath, filename));
+		assertNotEquals(value2LastModified, value1LastModified);
+		
+		Thread.sleep(1);
+		
+		TestValue value3 = new TestValue("Leroy Jenkins", 0);
+		fileManager.saveVersionedObject(tempDirPath, filename, value3);
+		
+		Path value3Path = FileManager.getNewestVersionPath(tempDirPath, filename);
+		FileTime value3LastModified = Files.getLastModifiedTime(value3Path);
+		
+		assertEquals(value1, fileManager.loadObject(value1Path));
+		assertEquals(value1LastModified, Files.getLastModifiedTime(value1Path));
+		assertEquals(value2, fileManager.loadObject(value2Path));
+		assertEquals(value2LastModified, Files.getLastModifiedTime(value2Path));
+		assertEquals(value3, fileManager.loadVersionedObject(tempDirPath, filename));
+		assertNotEquals(value3LastModified, value2LastModified);
+	}
+	
+	@Test
+	public void test_getNewestVersionPath() throws IOException {
+		File tempDir = createTempFolder("test_getNewestVersionPath");
+		Path tempDirPath = tempDir.toPath();
+		String filename = "test_filename";
+		
+		Random r = new Random();
+		SimpleDateFormat sdf = new SimpleDateFormat(FileManager.TIMESTAMP_VERSION_FORMAT);
+		
+		assertNull(FileManager.getNewestVersionPath(tempDirPath, filename));
+		
+		List<Path> testPaths = new LinkedList<Path>();
+		
+		Path legacyPath = tempDirPath.resolve(filename);
+		Files.createFile(legacyPath);
```

Comment:
Before you add  the timestamped files, do you want to check that "assertEquals(legacyPath, FileManager.getNewestVersionPath(tempDirPath, filename));" ?

Reply:
I am covering that case in the `test_saveAndLoadVersionedObjects` method, but I think you're right that it makes sense here as well. I just pushed a change for this.